OS X build broken due triggered warnings

The OS X build fails with the following error:

    [ 43%] Building CXX object src/Entities/CMakeFiles/Entities.dir/Player.cpp.o
    /Users/matti/hacking/cuberite/src/Entities/Player.cpp:942:11: warning: enumeration value 'dtAttack' not explicitly handled in switch [-Wswitch-enum]
                    switch (a_TDI.DamageType)
                            ^
    /Users/matti/hacking/cuberite/src/Entities/Player.cpp:2127:50: error: using integer absolute value function 'abs' when argument is of floating point type [-Werror,-Wabsolute-value]
                                    m_Stats.AddValue(statDistFallen, (StatValue)(abs(a_DeltaPos.y) * 100 + 0.5));
                                                                                 ^
    /Users/matti/hacking/cuberite/src/Entities/Player.cpp:2127:50: note: use function 'std::abs' instead
                                    m_Stats.AddValue(statDistFallen, (StatValue)(abs(a_DeltaPos.y) * 100 + 0.5));
                                                                                 ^~~
                                                                                 std::abs
    /Users/matti/hacking/cuberite/src/Entities/Player.cpp:2127:38: error: use of old-style cast [-Werror,-Wold-style-cast]
                                    m_Stats.AddValue(statDistFallen, (StatValue)(abs(a_DeltaPos.y) * 100 + 0.5));
                                                                     ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/matti/hacking/cuberite/src/Entities/Player.cpp:2127:65: warning: implicit conversion turns floating-point number into integer: 'const double' to 'int' [-Wfloat-conversion]
                                    m_Stats.AddValue(statDistFallen, (StatValue)(abs(a_DeltaPos.y) * 100 + 0.5));
                                                                                 ~~~ ~~~~~~~~~~~^
    /Users/matti/hacking/cuberite/src/Entities/Player.cpp:2141:13: warning: 30 enumeration values not explicitly handled in switch: 'mtInvalidType', 'mtCreeper', 'mtSkeleton'... [-Wswitch-enum]
                                    switch (Monster->GetMobType())
                                            ^
    /Users/matti/hacking/cuberite/src/Entities/Player.cpp:2134:11: warning: 11 enumeration values not explicitly handled in switch: 'etEntity', 'etEnderCrystal', 'etPlayer'... [-Wswitch-enum]
                    switch (m_AttachedTo->GetEntityType())
                            ^
    4 warnings and 2 errors generated.
    src/Entities/CMakeFiles/Entities.dir/build.make:494: recipe for target 'src/Entities/CMakeFiles/Entities.dir/Player.cpp.o' failed
    make[2]: *** [src/Entities/CMakeFiles/Entities.dir/Player.cpp.o] Error 1
    CMakeFiles/Makefile2:1657: recipe for target 'src/Entities/CMakeFiles/Entities.dir/all' failed
    make[1]: *** [src/Entities/CMakeFiles/Entities.dir/all] Error 2
    Makefile:127: recipe for target 'all' failed
    make: *** [all] Error 2

The compiler in question is:

    Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
    Target: x86_64-apple-darwin14.4.0
    Thread model: posix

The code triggers two warnings (`absolute-value` and `old-style-cast`) that are promoted into errors in the build. In the former case the `abs` function resolves to the [`cstdlib`](http://en.cppreference.com/w/c/numeric/math/abs) version instead of the [`cmath`](http://en.cppreference.com/w/cpp/numeric/math/fabs) version which probably was the origin intention.

--- a/src/Entities/Player.cpp
+++ b/src/Entities/Player.cpp
@@ -2124,7 +2124,7 @@ void cPlayer::UpdateMovementStats(const Vector3d & a_DeltaPos, bool a_PreviousIs
 			else if (a_DeltaPos.y < 0.0)
 			{
 				// Increment statistic
-				m_Stats.AddValue(statDistFallen, (StatValue)(abs(a_DeltaPos.y) * 100 + 0.5));
+				m_Stats.AddValue(statDistFallen, static_cast<StatValue>(std::abs(a_DeltaPos.y) * 100 + 0.5));
 			}
 			// TODO: good opportunity to detect illegal flight (check for falling tho)
 		}

